se_services/zephyr/src/se_service.c

```diff
@@ -159,6 +159,9 @@ static int send_msg_to_se(uint32_t *ptr, uint32_t size, uint32_t timeout)
 	if (k_can_yield()) {
 		int wait = 0;
 
+		k_sem_reset(&svc_send_sem);
```

Comment:
I agree. First I was confused a little a bit. Maybe we could review data at callback_for_receive_msg() that it match request before give semaphore.